packages/backend/src/controllers/runQueryController.ts

```diff
@@ -0,0 +1,177 @@
+import { allowApiKeyAuthentication, isAuthenticated } from './authentication';
+
+type ApiRunQueryResponse = {
+    status: 'ok';
+    results: {
+        metricQuery: any; // tsoa doesn't support complex types like MetricQuery
+        rows: any[];
+    };
+};
+
+type RunQueryRequest = {
+    // tsoa doesn't support complex types like MetricQuery
+    dimensions: any[]; // Dimensions to group by in the explore
+    metrics: any[]; // Metrics to compute in the explore
+    filters: any;
+    sorts: any[]; // Sorts for the data
+    limit: number; // Max number of rows to return from query
+    tableCalculations: any[]; // calculations to append to results
+    additionalMetrics?: any[]; // existing metric type
+    csvLimit?: number;
+};
+
+@Route('/api/v1/projects/{projectUuid}')
+@Response<ApiErrorPayload>('default', 'Error')
+export class RunViewChartQueryController extends Controller {
+    /**
+     * Run a query for a chart on view mode
+     * @param projectUuid The uuid of the project
+     * @param body chartUuid for the chart to run
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Post('/runViewChartQuery')
+    @OperationId('postRunViewChartQuery')
+    async postViewChart(
+        @Body() body: { chartUuid: string },
+        @Path() projectUuid: string,
+        @Request() req: express.Request,
+    ): Promise<ApiRunQueryResponse> {
+        this.setStatus(200);
+        return {
+            status: 'ok',
+            results: await projectService.runViewChartQuery(
+                req.user!,
+                body.chartUuid,
+                projectUuid,
+            ),
+        };
+    }
+
+    /**
+     * Run a query for a chart from dashboards
+     * @param projectUuid The uuid of the project
+     * @param body chartUuid for the chart to run
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Post('/runDashboardTileQuery')
+    @OperationId('postRunDashboardTileQuery')
+    async postDashboardTile(
+        @Body() body: { chartUuid: string },
+        @Path() projectUuid: string,
+        @Request() req: express.Request,
+    ): Promise<ApiRunQueryResponse> {
+        this.setStatus(200);
+        return {
+            status: 'ok',
+            results: await projectService.runViewChartQuery(
+                req.user!,
+                body.chartUuid,
+                projectUuid,
+            ),
+        };
+    }
+
+    /**
+     * Run a query for underlying data results
+     * @param projectUuid The uuid of the project
+     * @param body metricQuery for the chart to run
+     * @param exploreId table name
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Post('/explores/{exploreId}/runUnderlyingDataQuery')
```

Comment:
This behaves the same as the runQuery endpoint? I think that's ok since permissions for them are the same, we can change that later if we need